statics/workflows/ping-mesh.yaml

```diff
@@ -0,0 +1,111 @@
+      var random = Math.floor(Math.random() * 65535);
+    } else {
+      var random = 1024 + Math.floor(Math.random() * (65535-1024));
+    }
+    return client.captures.create(capture).then(function (c) {
+      capture = c
+    }).then(function () {
+      sources.forEach(function(source) {
+        packetInjection.Src = "G.V().Has('TID', '" + source + "')"
+        packetInjection.Dst = "G.V().Has('TID', '" + to + "')"
+        packetInjection.Count = 5
+        return client.G.V().Has("TID", from).then(
+          function (nodes) {
+          if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+            packetInjection.SrcIP = nodes[0].Metadata.Neutron.IPV4[0]
+          }
+          if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+            packetInjection.SrcMAC = nodes[0].Metadata.ExtID["attached-mac"]
+          }
+          if (protocol == "ICMPv4") {
+            packetInjection.Type = "icmp4"
+            packetInjection.ICMPID = random + i + Math.floor(Math.random() * 1000);
+            i +=  Math.floor(Math.random() * 1000);
+            From[source] = packetInjection.ICMPID;
+          }
+          if (protocol == "TCP") {
+            packetInjection.Type = "tcp4"
+            packetInjection.SrcPort = random + i;
+            i +=  Math.floor(Math.random() * 1000);
+            packetInjection.DstPort = random + i;
+            i +=  Math.floor(Math.random() * 1000);
+            From[source] = [packetInjection.SrcPort, packetInjection.DstPort];
+          }
+          if (protocol == "UDP") {
+            packetInjection.Type = "udp4"
+            packetInjection.SrcPort = random + i;
+            i +=  Math.floor(Math.random() * 1000);
+            packetInjection.DstPort = random + i;
+            i +=  Math.floor(Math.random() * 1000);
+            From[source] = [packetInjection.SrcPort, packetInjection.DstPort];
+          }
+          i += 1;
+        }).then(function () {
+          return client.G.V().Has("TID", to)
+        }).then(function (nodes) {
+          if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+            packetInjection.DstIP = nodes[0].Metadata.Neutron.IPV4[0]
+          }
+          if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+            packetInjection.DstMAC = nodes[0].Metadata.ExtID["attached-mac"]
+          }
+          return client.packetInjections.create(packetInjection)
+        })
+      });
+    }).then(function () {
+      return sleep(2000)
+    }).then(function () {
+      sources.forEach(function(source) {
+        if (protocol == "ICMPv4") {
+          return client.G.Flows().Has("ICMP.ID", From[source]).then(
+          function (flows) {
+            dict[source] = flows.length > 0;
+          })
+        }
+        if (protocol == "TCP" || "UDP") {
+          return client.G.Flows().Has("Transport.A", From[source][0], "Transport.B", From[source][1]).then(
```

Comment:
check if the transport protocol is tcp or udp too

Reply:
++